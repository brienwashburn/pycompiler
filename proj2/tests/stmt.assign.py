a = b
a, b = c
a, = c
a, b, = c
a = yield
a = yield from baz
a, b, c = yield
a, b, c = yield from baz
a = foo(10 * f + 22)
a, b, *c = [1, 2, 3, 4, 5]
*a, *b, *foo, *x = 12
*_ = 12

a = b = 3
a = *b = c = 3*x
a = *b = c = d = *f = e = 10
*a = b = foo(3)[21]
a,b = *c = d = 3
*c = a,b = c,d,e = 10

a = b or c, d or e
a,b = c = d = e or f, g or h
