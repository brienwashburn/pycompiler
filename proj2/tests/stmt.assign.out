(Module
 (Assign (targets (Name a)) (value (Name b)))
 (Assign (targets (Tuple (Name a) (Name b))) (value (Name c)))
 (Assign (targets (Tuple (Name a))) (value (Name c)))
 (Assign (targets (Tuple (Name a) (Name b))) (value (Name c)))
 (Assign (targets (Name a)) (value (Yield)))
 (Assign (targets (Name a)) (value (YieldFrom (Name baz))))
 (Assign (targets (Tuple (Name a) (Name b) (Name c))) (value (Yield)))
 (Assign
  (targets (Tuple (Name a) (Name b) (Name c)))
  (value (YieldFrom (Name baz))))
 (Assign
  (targets (Name a))
  (value
   (Call
    (func (Name foo))
    (args (BinOp (BinOp (Num 10) Mult (Name f)) Add (Num 22)))
    (keywords)
    (starargs #f)
    (kwargs #f))))
 (Assign
  (targets (Tuple (Name a) (Name b) (Starred (Name c))))
  (value (List (Num 1) (Num 2) (Num 3) (Num 4) (Num 5))))
 (Assign
  (targets
   (Tuple
    (Starred (Name a))
    (Starred (Name b))
    (Starred (Name foo))
    (Starred (Name x))))
  (value (Num 12)))
 (Assign (targets (Starred (Name _))) (value (Num 12)))
 (Assign (targets (Name a) (Name b)) (value (Num 3)))
 (Assign
  (targets (Name a) (Starred (Name b)) (Name c))
  (value (BinOp (Num 3) Mult (Name x))))
 (Assign
  (targets
   (Name a)
   (Starred (Name b))
   (Name c)
   (Name d)
   (Starred (Name f))
   (Name e))
  (value (Num 10)))
 (Assign
  (targets (Starred (Name a)) (Name b))
  (value
   (Subscript
    (Call
     (func (Name foo))
     (args (Num 3))
     (keywords)
     (starargs #f)
     (kwargs #f))
    (Index (Num 21)))))
 (Assign
  (targets (Tuple (Name a) (Name b)) (Starred (Name c)) (Name d))
  (value (Num 3)))
 (Assign
  (targets
   (Starred (Name c))
   (Tuple (Name a) (Name b))
   (Tuple (Name c) (Name d) (Name e)))
  (value (Num 10)))
 (Assign
  (targets (Name a))
  (value (Tuple (BoolOp Or (Name b) (Name c)) (BoolOp Or (Name d) (Name e)))))
 (Assign
  (targets (Tuple (Name a) (Name b)) (Name c) (Name d))
  (value (Tuple (BoolOp Or (Name e) (Name f)) (BoolOp Or (Name g) (Name h))))))
